lib/Sema/CSSimplify.cpp

```diff
@@ -3525,14 +3525,40 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
   if (shouldAttemptFixes() && baseObjTy->getOptionalObjectType()) {
     // If the base type was an optional, look through it.
 
-    // We're unwrapping the base to perform a member access.
-    if (recordFix(Fix::getUnwrapOptionalBase(*this, member), locator))
-      return SolutionKind::Error;
-    
+    // If the base type is optional because we haven't chosen to force an
+    // implicit optional, don't try to fix it. The IUO will be forced instead.
+    auto resolvedOverload = getResolvedOverloadSets();
+    while (resolvedOverload) {
+      if (resolvedOverload->BoundType->isEqual(originalBaseTy)) {
```

Comment:
I don't think that checking `BoundType` for equality with `originalBaseTy` will suffice; it could match for unrelated cases. Could you check the locators here instead, like you're doing in `CSApply.cpp`?